src/panels/lovelace/views/hui-panel-view.ts

```diff
@@ -1,93 +1,113 @@
-  UpdatingElement,
+  TemplateResult,
 } from "lit-element";
-import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
-import { LovelaceViewConfig } from "../../../data/lovelace";
-import { HomeAssistant } from "../../../types";
-import { createCardElement } from "../create-element/create-card-element";
-import { Lovelace, LovelaceCard } from "../types";
+import { classMap } from "lit-html/directives/class-map";
+import { computeRTL } from "../../../common/util/compute_rtl";
+import type {
+  LovelaceViewConfig,
+  LovelaceViewElement,
+} from "../../../data/lovelace";
+import type { HomeAssistant } from "../../../types";
+import { HuiErrorCard } from "../cards/hui-error-card";
+import { showEditCardDialog } from "../editor/card-editor/show-edit-card-dialog";
+import type { Lovelace, LovelaceCard } from "../types";
 
 let editCodeLoaded = false;
 
-@customElement("hui-panel-view")
-export class HUIPanelView extends UpdatingElement {
-  @property({ attribute: false }) public hass?: HomeAssistant;
+export class PanelView extends LitElement implements LovelaceViewElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property({ attribute: false }) public lovelace?: Lovelace;
 
-  @property() public config?: LovelaceViewConfig;
+  @property({ type: Number }) public index?: number;
 
-  @property({ type: Number }) public index!: number;
+  @property({ type: Number }) public columns!: number;
 
-  protected firstUpdated(changedProperties: PropertyValues): void {
-    super.firstUpdated(changedProperties);
+  @property({ attribute: false }) public cards: Array<
+    LovelaceCard | HuiErrorCard
+  > = [];
+
+  @property({ type: Boolean }) public editMode = false;
+
+  public constructor() {
+    super();
     this.style.setProperty("background", "var(--lovelace-background)");
   }
 
-  protected update(changedProperties: PropertyValues): void {
-    super.update(changedProperties);
+  public setConfig(_config: LovelaceViewConfig): void {}
 
-    const hass = this.hass!;
-    const lovelace = this.lovelace!;
-    const hassChanged = changedProperties.has("hass");
-    const oldHass = changedProperties.get("hass") as this["hass"] | undefined;
-    const configChanged = changedProperties.has("config");
+  protected updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
 
-    if (lovelace.editMode && !editCodeLoaded) {
+    if (this.editMode && !editCodeLoaded) {
       editCodeLoaded = true;
-      import(/* webpackChunkName: "hui-view-editable" */ "./hui-view-editable");
+      import(
+        /* webpackChunkName: "default-layout-editable" */ "./default-view-editable"
+      );
     }
 
-    let editModeChanged = false;
-
-    if (changedProperties.has("lovelace")) {
-      const oldLovelace = changedProperties.get("lovelace") as Lovelace;
-      editModeChanged =
-        !oldLovelace || lovelace.editMode !== oldLovelace.editMode;
-    }
+    this._createCard();
```

Comment:
We are now recreating the card on every state change?

Reply:
And we no longer set the theme

Reply:
And we need to update `hass` if it is changed on the card.
```ts
this._card.hass = this.hass;
```

Reply:
Updating hass happens in the `hui-view`

Reply:
The theme is currently being applied in `hui-view` as well. Do we need to apply it in each separate view instead?

Reply:
We still call this function on every state change? Should then only be called when edit mode is changed and the first time?